compiler/src/org.graalvm.compiler.core.aarch64.test/src/org/graalvm/compiler/core/aarch64/test/AArch64ConditionalSetTest.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2018, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2018, Arm Limited and affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package org.graalvm.compiler.core.aarch64.test;
+
+import jdk.vm.ci.aarch64.AArch64;
+import org.graalvm.compiler.core.common.calc.UnsignedMath;
+import org.graalvm.compiler.core.test.GraalCompilerTest;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assume.assumeTrue;
+
+public class AArch64ConditionalSetTest extends GraalCompilerTest {
+    @Before
+    public void checkAArch64() {
+        assumeTrue("skipping AArch64 specific test", getTarget().arch instanceof AArch64);
+    }
+
+    /**
+     * conditionalSet test for integer tests condition.
+     */
+    public static int conditionalSetEQZero(int m) {
+        if ((m & 2) == 0) {
+            return 1;
+        }
+        return 0;
+    }
+
+    @Test
+    public void testConditionalSetEQZero() {
+        test("conditionalSetEQZero", 0);
```

Comment:
This testclass only checks if the result is correct but does not verify if the proper LIR instruction was selected.

Can we do the same as we do in the MatchRuleTests with `MatchRuleTest.checkLIR`, so that we can actually verify, that the LIR instruction was applied. The simplest way is to put this class as a subclass of `AArch64MatchRuleTest` even though it is not testing a match rule. This activates the guard to test only for AArch64 and provides the `checkLIR` method.

Reply:
Yes, make sense! I just thought it was not testing a match rule. I will modify this like what you suggest. Thanks!